go/plugins/dotprompt/dotprompt.go

```diff
@@ -321,3 +329,99 @@ func sortSchemaSlices(s *jsonschema.Schema) {
 		sortSchemaSlices(s.Items)
 	}
 }
+
+// Adds a variant name to the prompt
+func WithVariant(variant string) PromptOption {
+	return func(p *Prompt) error {
+		p.Config.Variant = variant
+		return nil
+	}
+}
+
+// Adds tools to the prompt
+func WithTools(tools ...ai.Tool) PromptOption {
+	return func(p *Prompt) error {
+		var toolSlice []ai.Tool
+		toolSlice = append(toolSlice, tools...)
+		p.Tools = toolSlice
+		return nil
+	}
+}
+
+// Adds Generation details for the model to the prompt.
+func WithGenerationConfig(config *ai.GenerationCommonConfig) PromptOption {
+	return func(p *Prompt) error {
+		p.Config.GenerationConfig = config
+		return nil
+	}
+}
+
+// Adds the struct for input to the prompt.
```

Comment:
It's not necessarily a struct, it could be a primitive too I think. Maybe "WithInputType uses the type provided to derive the input schema." Same for output.